app/routes/analytics.js

```diff
@@ -12,9 +12,9 @@ router.post('/fingerprint', async (req, res) => {
         const hashedFingerprint = hashFingerprint(fingerprint)
         const route = await Counter.findOne({url: currentUrl})
         if(!route) {
-            const articleId = currentUrl.substr(10)
-            if(!ObjectId.isValid(articleId)) return res.sendStatus(500)      
-            await Counter.create({ url: currentUrl, viewCount: 1, visitedUsers: [hashedFingerprint], articleId})
+            const articleLink = currentUrl.substr(10)
+            if(!articleLink) return res.sendStatus(500)      
+            await Counter.create({ url: currentUrl, viewCount: 1, visitedUsers: [hashedFingerprint], articleLink: encodeURIComponent(articleLink)})
```

Comment:
`articleLink: encodeURIComponent(articleLink)`
Isn't articleLink requiring first replacing spaces with dashes? In case if it's already a properly encoded URL, why encode it second time? Did you check whether counter still works?

Reply:
Counter functions as intended, I'll review the code and answer the other part shortly

Reply:
Alright, the encoding is occurring here in the off chance that someone's fullname in the future has some strange character that becomes altered someway, as such I perform the encoding before comparison to prevent some weird case where it's not finding the name that it should.